src/azul/types.py

```diff
@@ -125,6 +146,21 @@ def json_elements_are_dicts(vs: MutableJSONArray) -> TypeGuard[MutableJSONs]:
     return True
 
 
+def json_mapping_of_dicts(vs: MutableJSON) -> Mapping[str, MutableJSON]:
+    assert json_values_are_dicts(vs)
+    return vs
+
+
+def json_values_are_dicts(vs: MutableJSON) -> TypeGuard[Mapping[str, MutableJSON]]:
```

Comment:
L, but more importantly

```suggestion
def json_items_are_dicts(vs: MutableJSON) -> TypeGuard[dict[str, MutableJSON]]:
```

which also addresses the L